scripts/linters/pylint_extensions.py

```diff
@@ -2919,6 +2919,55 @@ def visit_importfrom(self, node: astroid.nodes.ImportFrom) -> None:
                 self.add_message('disallowed-text-import', node=node)
 
 
+# TODO(#16567): Here we use MyPy ignore because the incomplete typing of
+# pylint library and absences of stubs in pylint, forces MyPy to
+# assume that BaseChecker class has attributes of type Any.
+# Thus to avoid MyPy's error
+# (Class cannot subclass 'BaseChecker' (has type 'Any')),
+# we added an ignore here.
+class PreventStringConcatenationChecker(checkers.BaseChecker): # type: ignore[misc]
+    """Checks for string concactenation and encourages string interpolation."""
+
+    __implements__ = interfaces.IAstroidChecker
+
+    name = 'use-string-interpolation'
+    priority = -1
+    msgs = {
+        'C0041': (
+            'Please use string interpolation instead of string concatenation',
+            'use-string-interpolation',
+            'Used when string concatenation is detected.',
+        ),
+    }
+
+    def visit_binop(self, node: astroid.BinOp) -> None:
+        """Visits a binary operation node in the AST
+        then checks if it's a string concatenation operation.
+
+        Args: node: astroid.BinOp The binary operation node to check.
+        """
+
+        if isinstance(node, astroid.BinOp) and node.op == '+':
+            try:
+                left_inferred = next(node.left.infer())
+                right_inferred = next(node.right.infer())
+            except astroid.exceptions.InferenceError:
+                return
+
+            if any(isinstance(inferred, (astroid.Instance, astroid.Const)) and
+                    isinstance(inferred.pytype(), str) and
+                    'datetime.datetime' in inferred.pytype() 
+                    for inferred in [left_inferred, right_inferred]):
```

Comment:
Can you leave a comment explaining what case this is designed to handle? Also please fix the style:

```suggestion
            if any(
                isinstance(inferred, (astroid.Instance, astroid.Const)) and
                isinstance(inferred.pytype(), str) and
                'datetime.datetime' in inferred.pytype() 
                for inferred in [left_inferred, right_inferred]
            ):
```

The above change fixes 2 things:
1. When code inside brackets/parentheses needs to be split multiple lines, the whole contents of the brackets/parentheses should be indented in a single block rather than treating the first line specially and only indenting the later lines. This is enforced by a linter for lists, but the same principle applies here.
2. Follows the same indentation pattern as the next `if` statement for long `if` conditions

Reply:
Done.